File: python/space-age/space_age.py
def secondToYear(seconds):
    return seconds / 31557600.0

class SpaceAge:
    def __init__(self, seconds):
        self.age = seconds

    def on_earth(self):
        result = secondToYear(self.age) / 1.0
        return float(format(result, '.2f'))

    def on_mercury(self):
        result = secondToYear(self.age) / 0.2408467
        return float(format(result, '.2f'))

    def on_venus(self):
        result = secondToYear(self.age) / 0.61519726
        return float(format(result, '.2f'))

    def on_mars(self):
        result = secondToYear(self.age) / 1.8808158
        return float(format(result, '.2f'))

    def on_jupiter(self):
        result = secondToYear(self.age) / 11.862615
        return float(format(result, '.2f'))

    def on_saturn(self):
        result = secondToYear(self.age) / 29.447498
        return float(format(result, '.2f'))

    def on_uranus(self):
        result = secondToYear(self.age) / 84.016846
        return float(format(result, '.2f'))

    def on_neptune(self):
        result = secondToYear(self.age) / 164.79132
        return float(format(result, '.2f'))

File: ruby/space-age/space_age.rb
class SpaceAge
  def initialize(seconds)
    @seconds = seconds
  end

  PLANETS = { 
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
  }

  def on_earth
    @seconds / 60 / 60 / 24 / 365.25
  end

  def on_mercury
    on_earth / PLANETS[:mercury]
  end

  def on_venus
    on_earth / PLANETS[:venus]
  end
  
  def on_mars
    on_earth / PLANETS[:mars]
  end
  
  def on_jupiter
    on_earth / PLANETS[:jupiter]
  end

  def on_saturn
    on_earth / PLANETS[:saturn]
  end

  def on_uranus
    on_earth / PLANETS[:uranus]
  end

  def on_neptune
    on_earth / PLANETS[:neptune]
  end
end
